### ISSUE TITLE
damo status failure

### ISSUE THREAD

**Issue by honggyukim** [awslabs/damo#103]
I've started damo with 78fc0606db35e2e7da2c16ba51eaab5107ad8b0f, then ran `damo status` but saw the following error.
```
$ sudo ./damo status
Traceback (most recent call last):
  File "/home/honggyu/work/damo/./damo", line 116, in <module>
    main()
  File "/home/honggyu/work/damo/./damo", line 113, in main
    subcmd.execute(args)
  File "/home/honggyu/work/damo/src/_damo_subcmds.py", line 29, in execute
    self.module.main(args)
  File "/home/honggyu/work/damo/src/damo_status.py", line 97, in main
    kdamonds, err = _damon.update_read_kdamonds(
  File "/home/honggyu/work/damo/src/_damon.py", line 1257, in update_read_kdamonds
    err = update_schemes_status(update_stats, update_tried_regions,
  File "/home/honggyu/work/damo/src/_damon.py", line 1227, in update_schemes_status
    if quota_effective_bytes and feature_supported('schemes_quota_effective_bytes'):
  File "/home/honggyu/work/damo/src/_damon.py", line 1094, in feature_supported
    return _damon_fs.feature_supported(feature)
  File "/home/honggyu/work/damo/src/_damon_sysfs.py", line 665, in feature_supported
    return feature_supports[feature]
KeyError: 'schemes_quota_effective_bytes'
```
The tested kernel is v6.10-rc5 + migrate action patches from mm-stable.

**Comment by honggyukim** [comment]
We might need the following change.
```diff
diff --git a/src/_damon_sysfs.py b/src/_damon_sysfs.py
index 65ba900..df95e08 100644
--- a/src/_damon_sysfs.py
+++ b/src/_damon_sysfs.py
@@ -662,6 +662,8 @@ feature_supports = None
 def feature_supported(feature):
     if feature_supports == None:
         update_supported_features()
+    if feature not in feature_supports:
+        return False
     return feature_supports[feature]

 # sysfs was merged in v5.18-rc1
```
It makes `damo status` work fine.

**Comment by sj-aws** [comment]
Hi Honggyu,


Thank you for this kind issue report.  I confirmed this issue can be reproduced only in specific steps such as below.

1. damo start; damo stop
2. Remove item for `schemes_quota_effective_bytes` from `$HOME/.damo.damon_features_supports` file.
3. damo status

The first two steps can also be done by running `damo start` with a version of damo that doesn't aware `schemes_quota_effective_bytes` feature.

Based on that, I found root cause of the issue, made a fix[1], and confirmed it works with above explained repro.  I now pushed the fix to `next` branch.  Could you please further confirm if it works?

[1] commit 3fe00f8b79a1 ("_damon: Handle a case that feature_supports file is written by an old version")

**Comment by sj-aws** [comment]
Opening this issue again, since the fix commit has closed this automatically, but we didn't get the confirmation from Honggyu.

**Comment by sj-aws** [comment]
Please note that the fix was having another issue that makes `damo record onging` always fails.  Just made a fix for it: 81801252e244 ("_damon: Do feature supports cache completeness check from set_feature_supports()")

**Comment by honggyukim** [comment]
Hi SeongJae,

I've run it again after update and it shows
```
$ sudo ./damo status
staging feature check purpose kdamond failed
```
I guess the problem happens when the versions of `damo start` and `damo status` are different.

Thanks for the quick fix!
